File: internal/adapters/lsp/servers/tsls/testdata/basic/references.ts
import { FIXTURE_STAMP, FixtureBucket, makeBucket } from "./fixture";

export function useBucket(): string {
    const left = new FixtureBucket(FIXTURE_STAMP);
    const right = makeBucket("secondary");

    return `${left.describe()}|${right.describe()}`;
}

File: internal/adapters/lsp/servers/gopls/testdata/basic/references.go
package basic

// UseMakeBucketTwice gives integration tests one symbol with two references to MakeBucket.
func UseMakeBucketTwice(value string) string {
	left := MakeBucket(value)
	right := MakeBucket(left.Describe())

	return right.Describe()
}

// UseMakeBucketOnce gives integration tests one symbol with one reference to MakeBucket.
func UseMakeBucketOnce(value string) FixtureBucket[string] {
	return MakeBucket(value)
}

// UseMakeBucketPrivateTwice gives integration tests one symbol with two references to makeBucketPrivate.
func UseMakeBucketPrivateTwice(value string) string {
	left := makeBucketPrivate(value)
	right := makeBucketPrivate(left.Describe())

	return right.Describe()
}
